gapis/api/transform2/transform_chain.go

```diff
@@ -157,23 +181,27 @@ func (chain *TransformChain) GetNextTransformedCommands(ctx context.Context) err
 		log.I(ctx, "Transforming... (%v:%v)", chain.currentCommandID, currentCommand)
 	}
 
-	err := chain.transformCommand(ctx, currentCommand, chain.currentCommandID)
+	inputCmds := make([]api.Cmd, 0)
+	inputCmds = append(inputCmds, currentCommand)
+	err := chain.transformCommands(ctx, chain.currentCommandID, inputCmds, 0)
 	if err != nil {
 		log.E(ctx, "Replay error (%v:%v): %v", chain.currentCommandID, currentCommand, err)
 	}
+
 	chain.currentCommandID++
 	return err
 }
 
-func (chain *TransformChain) handleInitialState(state *api.GlobalState) (*api.GlobalState, error) {
```

Comment:
I already mentioned it above, but even though this function does nothing at the moment, I'd prefer to keep it because it makes a strong statement of intent about what will happen in the future and where it should happen.

Reply:
Well, I think we have talked about not copying the state because it is very expensive to do so. Do we really want to copy it in the future? I thought that we had dropped the idea completely.

Reply:
It's required for functions that NeedAccurateState(). Cloning is expensive, but we can't have accurate state without doing it. So we will need to implement this in the future.